### ISSUE TITLE
Accessibility: jupyter.org/hub

### ISSUE THREAD

**Issue by Abolade017** [jupyterhub/outreachy#74]
### What page is this for?

https://jupyter.org/hub

### WAVE accessibility report

I evaluated the web accessibility of [https://jupyter.org/hub](url) page. My evaluation is the following:

![jupyterhub](https://user-images.githubusercontent.com/70269933/195314030-2c7dcad0-5cf3-44db-8641-8865b34275aa.JPG)

### **Detailed Evaluation**

- **Missing first level heading element (1) :** A page does not have an `<h1>` element. Headings facilitate page navigation for users of many assistive technologies. They also provide semantic and visual meaning and structure to the document. A first level heading (`<h1>`) should be present on nearly all pages. It should contain the most important heading on the page (generally the document title). If the page presents a main heading, place it within an `<h1>` element. Add other sub-headings as necessary.
- **Alternative text (2) :** A non-empty alt attribute is present on an image. Alternative text presents the content or function of an image to screen reader users or in other situations where images cannot be seen or are unavailable. It can be fixed by ensuring that the alternative text conveys the content and function of the image accurately and succinctly. The `alt` attribute should be equivalent, accurate, and succinct.
- **Linked image with alternative text (1) :** An image element has non-empty alternative text, is within a link, and no other text (or images with alternative text) is present within the link. Including appropriate alternative text on an image within a link ensures that the function and purpose of the link and the content of the image is available to screen reader users or when images are unavailable. It can be fixed by ensuring that the alternative text presents the content of the image and/or the function of the link. If the full content and function of the link is presented in text within the link (an image and a text caption both within the same link, for example), then the image should generally be given empty/null alternative text (`alt=""`) to avoid redundancy.
- **Language (1) :** A document or an element has a valid `lang` attribute value. Identifying the language of the page or portion of page (if different from the page itself) allows screen readers to read the content appropriately. It can be fixed by ensuring the language is properly identified for the page (e.g., `<html lang="en">`). If content within the page is in a language different than the page's language, identify it using a valid `lang` attribute value (e.g., `<p lang="fr">`).
- **Heading level 2 (7) :** An `<h2>` element is present. Headings facilitate page navigation for users of assistive technologies. They also provide semantic and visual meaning and structure to the document. It can be fixed by ensuring that the text in question is truly a heading and that it is structured correctly in the page outline.
- **Unordered list (4) :** A `<ul>` element is present. It can be fixed by ensuring that an unordered (bulleted) list is appropriate for the context. If list items are sequential or numbered, an ordered list (`<ol>`) is likely more appropriate.
- **Navigation (1) :** A `<nav>` element or role="navigation" is present. The navigation identifies a section of navigation links and can facilitate page semantics and navigation. It can be fixed by ensuring that the element defines page navigation. Multiple navigation elements on one page can be differentiated with `ARIA` labels.
- **Footer (1) :** A `<footer>` element or `role="contentinfo"` is present. Footers identify a footer for the page or a page section. It typically identifies authorship, related links, copyright date, or other footer content. Footers facilitate page semantics and navigation. It can be fixed by ensuring that the element surrounds and defines page or page section footer content.
- **ARIA labels** An `aria-label` or `aria-labelledby` attribute is present. ARIA labels define accessible names to be read by screen readers for interface elements. They may be used when HTML associations (label, alternative text, etc.) are not sufficient. It can be fixed by ensuring the `aria-label` value or element referenced by `aria-labelledby` provides a descriptive, accurate label. When possible, use standard `HTML <label>` or other markup to make the association.

### The HTML element to be changed

```html
1. <nav class="navbar"><div class="navbar-header"><a class="navbar-brand" href="/"><picture><source media="(max-width: 767px)" srcset="/assets/logos/logomark-orangebody-greyplanets.svg"><source media="(min-width: 768px)" srcset="/assets/logos/rectanglelogo-greytext-orangebody-greymoons.svg">
Linked image with alternative text<img class="navbar-logo" src="/assets/logos/rectanglelogo-greytext-orangebody-greymoons.svg" alt="Jupyter Home" loading="eager" height="40">
</picture></a></div><button type="button" class="hamburger" data-bs-toggle="collapse" data-bs-target="#bs-navbar-collapse"><span class="sr-only">
Toggle navigation
</span><span class="icon-bar white-icon-bar"></span><span class="icon-bar white-icon-bar"></span><span class="icon-bar white-icon-bar"></span></button><div class="mobile-clearfix"></div><div class="collapse navbar-collapse" id="bs-navbar-collapse">
Unordered list<ul class="navbar-links"><li><a href="/try" class="">
Try
</a></li><li><a href="/install" class="">
Install
</a></li><li><a href="/community" class="">
Get Involved
</a></li><li><a href="https://docs.jupyter.org/" target="_blank" rel="noopener noreferrer" class="">
Documentation
</a></li><li><a href="https://blog.jupyter.org" target="_blank" rel="noopener noreferrer" class="">
News
</a></li><li><a href="/governance/" class="">
Governance
</a></li><li><a href="/security" class="">
Security
</a></li><li><a href="/about" class="">
About
</a></li></ul>
</div></nav> is not in a landmark
2. i will remove <header> element from <header class="page-header">
Alternative text<img src="assets/homepage/hublogo.svg" alt="JupyterHub logo" loading="eager">
<p>
A multi-user version of the notebook designed for companies, classrooms and research labs
</p><script src="https://cdn.jsdelivr.net/npm/anchor-js/anchor.min.js"></script><script>
document.addEventListener('DOMContentLoaded', function(event) { anchors.add(); });
</script></header>
2. The page is missing <h1> element
3. The aria-label attribute is having the same value 'Anchor' throughout the page.
4. The accessible name of <li><a href="https://voila.readthedocs.io/" target="_blank" rel="noopener noreferrer">Voilà</a></li> is not an English word and lang attributed is not set for it.
5. I will change the value given to the aria-label in <a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="#follow-us" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a> throughout the page.
```

### Your proposed HTML (or CSS), after change

```html
1. I will put the <nav class="navbar"><div class="navbar-header"><a class="navbar-brand" href="/"><picture><source media="(max-width: 767px)" srcset="/assets/logos/logomark-orangebody-greyplanets.svg"><source media="(min-width: 768px)" srcset="/assets/logos/rectanglelogo-greytext-orangebody-greymoons.svg">
Linked image with alternative text<img class="navbar-logo" src="/assets/logos/rectanglelogo-greytext-orangebody-greymoons.svg" alt="Jupyter Home" loading="eager" height="40">
</picture></a></div><button type="button" class="hamburger" data-bs-toggle="collapse" data-bs-target="#bs-navbar-collapse"><span class="sr-only">
Toggle navigation
</span><span class="icon-bar white-icon-bar"></span><span class="icon-bar white-icon-bar"></span><span class="icon-bar white-icon-bar"></span></button><div class="mobile-clearfix"></div><div class="collapse navbar-collapse" id="bs-navbar-collapse">
Unordered list<ul class="navbar-links"><li><a href="/try" class="">
Try
</a></li><li><a href="/install" class="">
Install
</a></li><li><a href="/community" class="">
Get Involved
</a></li><li><a href="https://docs.jupyter.org/" target="_blank" rel="noopener noreferrer" class="">
Documentation
</a></li><li><a href="https://blog.jupyter.org" target="_blank" rel="noopener noreferrer" class="">
News
</a></li><li><a href="/governance/" class="">
Governance
</a></li><li><a href="/security" class="">
Security
</a></li><li><a href="/about" class="">
About
</a></li></ul>
</div></nav> in the <header> 
2. I will put <img src="assets/homepage/hublogo.svg" alt="JupyterHub logo" loading="eager">
<p>
A multi-user version of the notebook designed for companies, classrooms and research labs
</p><script src="https://cdn.jsdelivr.net/npm/anchor-js/anchor.min.js"></script><script>
document.addEventListener('DOMContentLoaded', function(event) { anchors.add(); });
</script> in the <header> element as well
2. I will add a <h1> element to the page.
3. I will change the aria-label attribute to a value that best describes the function of the <a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="#follow-us" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a>.
4. I will put 'lang=fr' in <li><a href="https://voila.readthedocs.io/" target="_blank" rel="noopener noreferrer">
Voilà
</a></li>.
5. I give each of the <a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="#follow-us" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a> the well appropriate value to the aria-label.
```

**Comment by minrk** [comment]
Good suggestions! I did not know you could put `lang` on any HTML element instead of just the page!

**Comment by sgibson91** [comment]
Thank you for your contribution! Since the contribution period is now over, we will close this issue.
